src/Microsoft.ML.FastTree/Representation/TreeRegressor.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Trainers.FastTree.Internal;
+
+namespace Microsoft.ML.FastTree.Representation
+{
+    /// <summary>
+    /// A container class for exposing <see cref="RegressionTree"/>'s attributes to users.
+    /// This class should not be mutable, so it contains a lot of read-only members.
+    /// </summary>
+    public class TreeRegressor
+    {
+        /// <summary>
+        /// <see cref="TreeRegressor"/> is an immutable wrapper over <see cref="_tree"/> for exposing some tree's attribute to users.
+        /// </summary>
+        private readonly RegressionTree _tree;
+
+        /// <summary>
+        /// Sample labels from training data. <see cref="_leafSamples"/>[i] stores the labels falling into the i-th leaf.
+        /// </summary>
+        private readonly List<double[]> _leafSamples;
+        /// <summary>
+        /// Sample labels' weights from training data. <see cref="_leafSampleWeights"/>[i] stores the weights for labels falling into the i-th leaf.
```

Comment:
>falling [](start = 121, length = 7)

ensure comments fit on once screen (<80 chars/line)

Reply:
Summary comments we've typically tried to keep at around 120 or 150 actually, from other files. 80 is a bit too harsh. (If you tried to enforce this policy the result would look ridiculous.) For markdown though we do try to keep it properly formatted to a "console window" width.

Reply:
Done.

---
In reply to: [251959736](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251959736) [](ancestors = 251959736,251949011)